src/test/java/org/springframework/social/openidconnect/api/impl/HttpClientFactoryTest.java

```diff
@@ -0,0 +1,208 @@
+package org.springframework.social.openidconnect.api.impl;
+
+import junit.framework.Assert;
+
+import org.apache.http.conn.ClientConnectionManager;
+import org.apache.http.pool.PoolEntry;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.client.*;
+import org.springframework.social.openidconnect.HttpClientFactory;
+import org.springframework.social.openidconnect.PayPalAccessException;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Proxy;
+import java.net.URI;
+
+
+/**
+ * Tests whether TLS version parameter works.  Tested by running under jdk 1.6 and 1.7
+ */
+public class HttpClientFactoryTest {
+    @Rule
+    public final ExpectedException exception = ExpectedException.none();
+
+    //private static final String TLSVER = "jdk.tls.client.protocols";
+    private static final String TLSVER = "tlsProtocol";
+
+    private static final double javaVersion = Double.parseDouble(System.getProperty("java.specification.version"));
+
+    @After
+    public void afterEach() {
+        System.clearProperty(TLSVER);
+    }
+
+    /**
+     * Test reading/writing TLS version
+     */
+    @Test
+    public void testJdkTlsProperty() {
+        String tlsProtocols = System.getProperty(TLSVER);
+        Assert.assertNull(tlsProtocols);
+
+        System.setProperty(TLSVER, "TLSv1.2");
+        tlsProtocols = System.getProperty(TLSVER);
+        Assert.assertEquals("TLSv1.2", tlsProtocols);
```

Comment:
I don't think we would like to test System.setProperty and System.getProperty is working or not